File: problems/2670-find-the-distinct-difference-array/solution.java
import java.util.HashSet;
import java.util.Set;

class Solution {
    public int[] distinctDifferenceArray(int[] nums) {
        int n = nums.length;
        int[] diff = new int[n];

        for (int i = 0; i < n; i++) {
            Set<Integer> prefixDistinct = new HashSet<>();
            for (int j = 0; j <= i; j++) {
                prefixDistinct.add(nums[j]);
            }

            Set<Integer> suffixDistinct = new HashSet<>();
            for (int j = i + 1; j < n; j++) {
                suffixDistinct.add(nums[j]);
            }

            diff[i] = prefixDistinct.size() - suffixDistinct.size();
        }

        return diff;
    }
}

File: problems/2670-find-the-distinct-difference-array/solution.cpp
class Solution {
public:
    vector<int> distinctDifferenceArray(vector<int>& nums) {
        int n = nums.size();
        vector<int> diff(n);
        for (int i = 0; i < n; ++i) {
            unordered_set<int> prefix_distinct;
            for (int j = 0; j <= i; ++j) {
                prefix_distinct.insert(nums[j]);
            }
            unordered_set<int> suffix_distinct;
            for (int j = i + 1; j < n; ++j) {
                suffix_distinct.insert(nums[j]);
            }
            diff[i] = prefix_distinct.size() - suffix_distinct.size();
        }
        return diff;
    }
};